demos/CMakeLists.txt

```diff
@@ -10,16 +10,28 @@ endif()
 # This function will add a demo target to the list of default targets
 # if all AWS Credentials are defined.
 function(check_aws_credentials demo_name)
-    if(AWS_IOT_ENDPOINT AND CLIENT_CERT_PATH AND CLIENT_PRIVATE_KEY_PATH)
-        target_compile_definitions(
-            ${demo_name} PRIVATE
-                AWS_IOT_ENDPOINT="${AWS_IOT_ENDPOINT}"
-                CLIENT_CERT_PATH="${CLIENT_CERT_PATH}"
-                CLIENT_PRIVATE_KEY_PATH="${CLIENT_PRIVATE_KEY_PATH}"
+    set(REQUIRED_AWS_DEFINES
+            "AWS_IOT_ENDPOINT"
+            "CLIENT_CERT_PATH"
+            "CLIENT_PRIVATE_KEY_PATH"
         )
-        set(CMAKE_REQUIRED_DEFINITIONS -DAWS_IOT_ENDPOINT -DCLIENT_CERT_PATH -DCLIENT_PRIVATE_KEY_PATH)
+    set(THING_NAME_DEMOS
+            "THING_NAME"
+        )
+    if(";${THING_NAME_DEMOS};" MATCHES ";${demo_name};")
```

Comment:
What is this step doing?

Reply:
These changes are unrelated to the PR's scope of adding the new automation job. Have created a separate PR #1457 instead to have a cleaner diff.